### PR TITLE
docs: update jsdoc examples (factory functions)

### PR DESCRIPTION
**Pull request by moniuch**
**Description:** jsdoc examples update in creation factories + converting result selectors to `map` + some linting

### CODE REVIEW

File: src/internal/observable/forkJoin.ts [hunk 1]
```diff
@@ -68,16 +68,16 @@ export function forkJoin<T>(...sources: ObservableInput<T>[]): Observable<T[]>;
  * when output Observable is supposed to emit a result.
  *
  * @example <caption>Use forkJoin with operator emitting immediately</caption>
- * import { forkJoin, of } from 'rxjs';
+ * import { forkJoin, of } from 'rxjs/observable';
```
Comment (cartant) [inline src/internal/observable/forkJoin.ts]: `import { forkJoin, of } from 'rxjs'`

File: src/internal/observable/forkJoin.ts [hunk 2]
```diff
@@ -86,17 +86,17 @@ export function forkJoin<T>(...sources: ObservableInput<T>[]): Observable<T[]>;
  *
  *
  * @example <caption>Use forkJoin with operator emitting after some time</caption>
- * import { forkJoin, interval } from 'rxjs';
+ * import { forkJoin, interval } from 'rxjs/observable';
```
Comment (cartant) [inline src/internal/observable/forkJoin.ts]: `import { forkJoin, interval } from 'rxjs'`

File: src/internal/observable/forkJoin.ts [hunk 3]
```diff
@@ -105,18 +105,19 @@ export function forkJoin<T>(...sources: ObservableInput<T>[]): Observable<T[]>;
  *
  *
  * @example <caption>Use forkJoin with project function</caption>
- * import { jorkJoin, interval } from 'rxjs';
+ * import { jorkJoin, interval } from 'rxjs/observable';
```
Comment (cartant) [inline src/internal/observable/forkJoin.ts]: `import { forkJoin, interval } from 'rxjs'` and also `jorkJoin`!

File: src/internal/observable/forkJoin.ts [hunk 4]
```diff
@@ -105,18 +105,19 @@ export function forkJoin<T>(...sources: ObservableInput<T>[]): Observable<T[]>;
...
  * import { take } from 'rxjs/operators';
  *
  * const observable = forkJoin(
  *   interval(1000).pipe(take(3)), // emit 0, 1, 2 every second and complete
- *   interval(500).pipe(take(4)), // emit 0, 1, 2, 3 every half a second and complete
- *   (n, m) => n + m
+ *   interval(500).pipe(take(4)),  // emit 0, 1, 2, 3 every half a second and complete
+ * ).pipe(
+ *   map((n, m) => n + m),
```
Comment (cartant) [inline src/internal/observable/forkJoin.ts]: `map(([n, m]) => n + m),` as `forkJoin` emits an array

File: src/internal/observable/throwError.ts [hunk 5]
```diff
@@ -16,20 +16,20 @@ import { Subscriber } from '../Subscriber';
  * Observables, such as in a {@link mergeMap}.
  *
  * @example <caption>Emit the number 7, then emit an error.</caption>
- * import { throwError, concat, of } from 'rxjs/create';
+ * import { throwError, concat, of } from 'rxjs/observable';
```
Comment (cartant) [inline src/internal/observable/throwError.ts]: `import { throwError, concat, of } from 'rxjs'`

File: src/internal/observable/throwError.ts [hunk 6]
```diff
@@ -16,20 +16,20 @@ import { Subscriber } from '../Subscriber';
...
  *
  * const result = concat(of(7), throwError(new Error('oops!')));
  * result.subscribe(x => console.log(x), e => console.error(e));
  *
  * @example <caption>Map and flatten numbers to the sequence 'a', 'b', 'c', but throw an error for 13</caption>
- * import { throwError, interval, of } from 'rxjs/create';
+ * import { throwError, interval, of } from 'rxjs/observable';
```
Comment (cartant) [inline src/internal/observable/throwError.ts]: `import { throwError, interval, of } from 'rxjs'`

**Comment by moniuch** [comment]
Spotted an old syntax leftover in combineLatest - fixed.
Believe I have fixed your comments @cartant 

_although I must admit I don't quit like github when it comes to reviewing own updates in PR... _

**Comment by moniuch** [comment]
@cartant Thanks. BTW, I have difficult time figuring out where I am in the comments/fixes flow - github still shows that you requested changes even though I have fixed them and the checks passed fine. Is there a solid way to make sure I haven't reeally missed a comment?

chrome_2018-06-02_13-44-26

**Comment by cartant** [comment]
@moniuch I'm not sure of your workflow. It looks to me like you've modified the actual commits in your PR, as they are all listed as having been added *after* my request for changes.

Usually, when making changes, I create new commits for each change and then push the branch to GitHub to update the PR. Then, only the new commits show up after the request for changes.

**Comment by moniuch** [comment]
@cartant Yes, exactly - what I do is: I make fixes, I squash the fix commits together with a corresponding old commit during `git rebase -i` (by using `fixup` on a commit), and then `git push -f` This is how new commits get created. It might fool github UI but the commits retain their scope and the histoy looks kind of "natural". However, if during merge all the commits are squashed anyway, then it might make more sens to just append new commits instead with changes requested.

If you prefer that the fixes be just appended, then I can adjust my workflow.

**Comment by cartant** [comment]
@moniuch Yeah, they'll be squashed when the PR is merged, so I think it's best to add new commits for any requested changes that might need to be made in the future. It's much easier for the reviewer to verify that their requests have been acted upon.

**Comment by moniuch** [comment]
OK, note taken for the future - thanks :)

**Comment by benlesh** [comment]
@moniuch ... same problem.. This needs rebased, too. So sorry. Docs are an area of heavy work right now :)

**Comment by moniuch** [comment]
@benlesh np, checks in progress

**Comment by benlesh** [comment]
Woo! Awesome! Thanks, @moniuch!

**Pull request merged**